src/core/ledger/grainAllocation.js

```diff
@@ -197,6 +208,29 @@ function immediateReceipts(
   return identities.map(({id}, i) => ({id, amount: amounts[i]}));
 }
 
+/**
+ * Split a grain budget based on exponentially weighted recent
+ * cred.
+ */
+function recentReceipts(
+  budget: G.Grain,
+  identities: ProcessedIdentities
+): $ReadOnlyArray<GrainReceipt> {
+  const discount = 0.1;
+  const amounts = G.splitBudget(
+    budget,
+    identities.map((i) =>
+      i.cred.reduce(
+        (acc, cred, index) =>
+          acc + cred * (1 - discount) ** (identities.length - index),
+        0
+      )
+    )
+  );
```

Comment:
This seems correct, but I think the following implementation is slightly more straightforward:

```js
i.cred.reduce((acc, cred) => acc * (1-discount) + cred, 0)
```

i.e. we don't need to explicitly find the right power to raise each cred component to, we can just decay it as we scan.

I also recommend factoring this slightly different, as in:

```js
const computeDecayedCred = (i) => i.cred.reduce((acc, cred) => acc * (1-discount) + cred, 0)
const decayedCredPerIdentity = identities.map(computeDecayedCred)
const amounts = G.splitBudget(budget, decayedCredPerIdentity)
```

Reply:
Ooh pretty, yes I agree this is a lot better.